e2e/tests-dfx/build.bash

```diff
@@ -96,6 +96,14 @@ teardown() {
   assert_match "syntax error"
 }
 
+@test "build --check fails on invalid motoko" {
+  install_asset invalid
+  dfx_start
+  # --check does not need create --all
```

Comment:
```suggestion
```

`dfx build --check` needs not `dfx start`.
There are three other tests above also only run `dfx build` with `--check`.

Reply:
I'm going to add a test that still calls `dfx start` and `dfx canister create --all` and verifies `dfx build --check`.  The reason is that there are slightly different code paths involved.

An earlier commit on this PR broke `dfx build --check` when there are no canister ids, but not when canister ids are known.